src/js/client/editprefset-viewport.js

```diff
@@ -0,0 +1,1001 @@
+            func: "{that}.reRender",
+            excludeSource: ["init"]
+        }
+    }
+});
+
+/**
+ * Common term filter counts
+ *
+ * Returns the total number of common terms, and then the maximum
+ * number that is set in the NP Set. This is used for rendering
+ * information by the settings filters in the form:
+ *
+ * [[My Settings (3)]] [All Settings (225)]
+ *
+ * This provides the user with contextual knowledge before hiding
+ * or showing all settings. This information could also be used
+ * to perhaps only show the first 10 settings or something, and
+ * then expand the rest when the user decides it is appropriate.
+ *
+ * If the user has multiple contexts, this will return the total
+ * number of unique settings keys in the NP Set across all contexts.
+ *
+ * @param {Array} commonTermsSorted - List of our common terms, currently
+ * known as generic preferences.
+ * @param {Object} flatPrefs - Users current set of flat preferences.
+ * @return {Object} Object with keys `npset` and `all`, such as
+ *                  {
+ *                      npset: 3,
+ *                      all: 225
+ *                  }
+ */
+gpii.devpmt.calculateCommonTermUsageCounts = function (commonTermsSorted, flatPrefs) {
+    var all = commonTermsSorted.length;
+    var npsetKeys = {};
+    fluid.each(flatPrefs.contexts, function (i) {
+        fluid.each(i.preferences, function (j, jKey) {
+            if (jKey.startsWith("http://registry.gpii.net/common")) {
+                npsetKeys[jKey] = true;
+            }
+        });
+    });
+    var counts = {
+        all: all,
+        npset: Object.keys(npsetKeys).length
+    };
+    return counts;
+};
+
+/**
+ * Calculate any changes in the common term filter counts and
+ * update the model. See `gpii.devpmt.commonTermUsageCounts` for
+ * details on the data.
+ */
+gpii.devpmt.updateCommonTermUsageCounts = function (that, commonTermsSorted, flatPrefs) {
+    var counts = gpii.devpmt.calculateCommonTermUsageCounts(commonTermsSorted, flatPrefs);
+    that.applier.change("commonTermUsageCounts", counts);
+};
+
+/**
+ * Lookup Generic Preference Setting.
+ *
+ * Using the context and commonTerm lookup the generic preference
+ * value. If the preference set does not contain the value, we
+ * return `undefined` to make this known, as `undefined` is not
+ * a valid JSON value. Theoretically, and preference setting maybe
+ * set as `null`.
+ */
+gpii.devpmt.lookupGenericPrefValue = function (that, context, commonTerm) {
+    if (that.model.flatPrefs.contexts[context].preferences[commonTerm] !== undefined) {
+        return that.model.flatPrefs.contexts[context].preferences[commonTerm];
+    }
+    else {
+        return undefined;
+    }
+};
+
+/**
+ * Lookup Application Specific Setting. Similar to `lookupGenericPrefValue` lookups
+ * in regard to return values, but takes an extra argument for the product.
+ */
+gpii.devpmt.lookupProductPrefValue = function (that, context, product, settingTerm) {
```

Comment:
The first argument can take in "{that}.model.flatPrefs.contexts" rather than the entire "{that}". As said, reduce the scope of arguments whenever possible. This comment applies to the entire pull request.

Reply:
thanks, this has been updated, and I will continue to review arguments to scope down.